### ISSUE TITLE
Stream is not closed when the receiver is closed

### ISSUE THREAD

**Issue by mrmlnc** [mrmlnc/fast-glob#239]
### Environment

  * OS Version: Windows 10
  * Node.js Version: 12

### Actual behavior

The `fast-glob` package will read directories in the stream, even after the receiver is closed.

### Expected behavior

When the receiver is closed, the read stream for each of patterns must be closed too.

### Code sample

```js
const fg = require('fast-glob');

(async () => {
    const stream = fg.stream('../DefinitelyTyped/**/*.md');

    for await (const entry of stream) {
        console.dir(entry, { colors: true });

        return;
    }
})();
```

**Comment by mrmlnc** [comment]
Original issue:

https://github.com/mrmlnc/fast-glob/pull/238#issuecomment-551775059

**Comment by mrmlnc** [comment]
#### How to test it?

Just add `console.log` before `emit` to `destination` stream [here](https://github.com/mrmlnc/fast-glob/blob/24a29c5c582966a95a9a8cca89b154d101434305/src/providers/stream.ts#L20).

**Comment by mrmlnc** [comment]
Will be fixed with #242.
